### PR TITLE
modified userdomain with factory pattern method

### PR DESCRIPTION
**Pull request by anushamh**
added user domain with factory pattern method

### CODE REVIEW

File: tests/Factory/UserFactoryTest.php [hunk 1]
```diff
@@ -0,0 +1,29 @@
+<?php
+namespace Notes\Model;
+
+use Notes\Model\User as UserModel;
+
+use Notes\Factory\UserFactory as UserFactory;
+
+class UserFactoryTest extends \PHPUnit_Framework_TestCase
+{
+    public function testCanSetMultipleProperties()
```
Comment (amitheda) [inline tests/Factory/UserFactoryTest.php]: Change test case name and Use test cases name with  '_'  format.

File: tests/Factory/UserFactoryTest.php [hunk 2]
```diff
@@ -0,0 +1,29 @@
+<?php
+namespace Notes\Model;
+
+use Notes\Model\User as UserModel;
```
Comment (amitheda) [inline tests/Factory/UserFactoryTest.php]: Use only  namespace or use operator for using UserModel in your code.

File: tests/Factory/UserFactoryTest.php [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+        $userFactory = new UserFactory();
+        $userModel   = $userFactory->create($input);
+        $this->assertEquals('anusha', $userModel->getFirstName());
+        $this->assertEquals('hiremath', $userModel->getLastName());
+        $this->assertEquals('anusha@gmail.com', $userModel->getEmail());
+        $this->assertEquals('sfhsk1223', $userModel->getPassword());
+        $this->assertEquals('2014-10-31 20:59:59', $userModel->getCreatedOn());
+        
+    }
+}
```
Comment (amitheda) [inline tests/Factory/UserFactoryTest.php]: Add more test cases for read,update operation.

File: app/Domain/User.php [hunk 4]
```diff
@@ -11,6 +11,8 @@
 
 use Notes\PasswordValidation\PasswordValidator as PasswordValidator;
 
+use Notes\Factory\UserFactory as UserFactory;
```
Comment (kalehrishi) [inline app/Domain/User.php]: class name should be just User

File: app/Factory/UserFactory.php [hunk 5]
```diff
@@ -0,0 +1,47 @@
+<?php
+namespace Notes\Factory;
+
+use Notes\Model\User as UserModel;
+
+class UserFactory
```
Comment (kalehrishi) [inline app/Factory/UserFactory.php]: class  name should be User

File: app/Factory/UserFactory.php [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+    }
+    
+    
+    public function read($input)
+    {
+        $userModel = new UserModel();
+        $userModel->setId($input['id']);
+        return $userModel;
+    }
+    public function readByUsernameandPassword($input)
```
Comment (kalehrishi) [inline app/Factory/UserFactory.php]: readByUsernameAndPassword

**Pull request closed without merging**